pennylane/compiler/qjit_api.py

```diff
@@ -17,43 +17,53 @@
 
 
 def qjit(fn=None, *args, compiler="catalyst", **kwargs):  # pylint:disable=keyword-arg-before-vararg
-    """A just-in-time decorator for PennyLane and JAX programs.
+    """A just-in-time hybrid quantum-classical compiler for PennyLane programs.
+
+    This decorator enables both just-in-time and ahead-of-time compilation,
+    depending on the compiler package and whether function argument type hints
+    are provided.
```

Comment:
Have modified the top of the docstring to make it very clear what the function does.

in addition, the first line of the function is _super_ important, as this is what will appear in autosummary tables elsewhere! so you want it to be concise yet descriptive.

Reply:
Thanks @josh146! I'll make sure to follow this in the followup PRs.